src/ops/fuelup_check.rs

```diff
@@ -0,0 +1,146 @@
+use std::collections::HashMap;
+use std::io::Write;
+
+use crate::{
+    commands::check::CheckCommand,
+    component::SUPPORTED_PLUGINS,
+    config::Config,
+    fmt::{bold, with_color_maybe_bold},
+    toolchain::Toolchain,
+};
+use anyhow::Result;
+use semver::Version;
+use termcolor::Color;
+
+use crate::{component, download::DownloadCfg};
+
+fn check_plugin(toolchain: &Toolchain, plugin: &str, latest_version: &Version) -> Result<()> {
+    let plugin_executable = toolchain.path.join(&plugin);
+    match std::process::Command::new(&plugin_executable)
+        .arg("--version")
+        .output()
+    {
+        Ok(o) => {
+            let version = Version::parse(
+                String::from_utf8_lossy(&o.stdout)
+                    .split_whitespace()
+                    .collect::<Vec<&str>>()[1],
+            )?;
+
+            print!("    - ");
+            bold(|s| write!(s, "{}", plugin));
+            print!(" - ");
+            if &version == latest_version {
+                with_color_maybe_bold(|s| write!(s, "Up to date"), Color::Green, true);
+                println!(" : {}", version);
+            } else {
+                with_color_maybe_bold(|s| write!(s, "Update available"), Color::Yellow, true);
+                println!(" : {} -> {}", version, latest_version);
+            }
+        }
+        Err(e) => {
+            print!("    - ");
+            bold(|s| write!(s, "{}", plugin));
+            print!(" - ");
+            if plugin_executable.exists() {
+                println!("execution error - {}", e);
+            } else {
+                println!("not found");
+            }
+        }
+    }
+    Ok(())
+}
+
+pub fn check(command: CheckCommand) -> Result<()> {
+    let CheckCommand { verbose } = command;
+
+    let cfg = Config::from_env()?;
+    let toolchains = cfg.list_toolchains()?;
+    let mut latest_versions: HashMap<String, Version> = HashMap::new();
+    let components = Vec::from([component::FORC, component::FUEL_CORE, component::FUELUP]);
+
+    for component in components {
+        let download_cfg: DownloadCfg = DownloadCfg::new(component, None)?;
+        latest_versions.insert(component.to_string(), download_cfg.version);
+    }
+
+    for toolchain in toolchains {
+        if let Ok(toolchain) = Toolchain::from(&toolchain) {
```

Comment:
Can this ever be an error? If we expect it to always be OK, we can `expect`, but if it can ever be an error, we should handle it.

Reply:
Thanks for this catch. I used to store the `TargetTriple` within the `Toolchain` struct in one of the earlier commits and have since removed it - main reason being that custom toolchains that users create will probably not have a target triple out of convenience. Slipped my mind to remove this section that checks the Ok.